### ISSUE TITLE
Not all async related log lines supressed after ASP MVC nuget package upgrade

### ISSUE THREAD

**Issue by kieranbenton** [aelij/AsyncFriendlyStackTrace#12]
Not sure which package caused it - or even if the framework version is affecting this ('4.0.30319.42000') but I've just mass upgraded a lot of nuget packages to their latest versions today in an ASP MVC project and now my log lines are back to being much more verbose...

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Test.Services.Catalog.Models.Mappings.ProductMapper.<MapPrices>b__18_2(BaseOffer x) in C:\Source\test\src\Test.Services\Catalog\Models\Mappings\ProductMapper.cs:line 317
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Test.Services.Catalog.Models.Mappings.ProductMapper.MapPrices(BaseProduct product, IEnumerable`1 alls, IEnumerable`1 priceCats, IEnumerable`1 priceTypes) in C:\Source\test\src\Test.Services\Catalog\Models\Mappings\ProductMapper.cs:line 311
   at async test.Services.Catalog.Models.Mappings.ProductMapper.<>c__DisplayClass16_0.<MapAsync>b__6(?) in C:\Source\test\src\Test.Services\Catalog\Models\Mappings\ProductMapper.cs:line 137
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Test.Services.Catalog.Models.Mappings.ProductMapper.MapAsync(?) in C:\Source\test\src\Test.Services\Catalog\Models\Mappings\ProductMapper.cs:line 92
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Test.Services.Catalog.ProductService.<GetByIdsWithNoContextInternalAsync>b__10_0(?) in C:\Source\test\src\Test.Services\Catalog\ProductService.cs:line 261
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Test.Caching.ICacheExtensions.MultiGetAsync[](?) in C:\Source\test\src\Test\Caching\ICacheExtensions.cs:line 61
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

I'm 99% certain I never used to see `at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()` anywhere?

Let me know if you need any more details.

**Comment by aelij** [comment]
Hmm. Seems that on .NET Core 1.1 the `ExceptionDispatchInfo.Throw` method doesn't get inlined. I'll add it to my filtering logic.

**Comment by kieranbenton** [comment]
Actually not on .NET Core - this is on the full fat framework 4.6.1. Same principle though I should imagine!

**Comment by aelij** [comment]
Fixed in e7ad2c61d2d88f06bc5bbb873b5648069bff8d91 (published v1.5 to NuGet)
